libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1042,11 +1042,13 @@ string ABIFunctions::abiEncodingFunctionFunctionType(
 		return createFunction(functionName, [&]() {
 			return Whiskers(R"(
 				function <functionName>(addr, function_id, pos) {
+					addr, function_id := <convert>(addr, function_id)
 					mstore(pos, <combineExtFun>(addr, function_id))
 				}
 			)")
 			("functionName", functionName)
 			("combineExtFun", m_utils.combineExternalFunctionIdFunction())
+			("convert", m_utils.conversionFunction(_from, _to))
```

Comment:
Added a conversion function. Should we also have a conversion function in the else branch? It might need to involve splitting, then conversion, then joining. @chriseth

Reply:
Since we know the type, maybe you could create a special conversion function that accepts the function value in combined form?